src/Frontend/Core/Engine/Url.php

```diff
@@ -257,7 +257,7 @@ private function determineLanguage(string $queryString): string
     private function setLanguageCookie(string $language): void
     {
         try {
-            self::getContainer()->get('fork.cookie')->set('frontend_language', $language);
+            $this->getContainer()->get('fork.cookie')->set('frontend_language', $language);
```

Comment:
Actually, when we switch to Symfony 4 we need to get the container again from a static property... https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing